Tests/TuistGeneratorTests/Linter/TargetLinterTests.swift

```diff
@@ -71,6 +71,24 @@ final class TargetLinterTests: TuistUnitTestCase {
         XCTContainsLintingIssue(got, LintingIssue(reason: "The target \(target.name) doesn't contain source files.", severity: .warning))
     }
 
+    func test_lint_when_target_no_source_files_but_has_dependency() {
+        let target = Target.test(sources: [], dependencies: [
+            TargetDependency.sdk(name: "libc++.tbd", status: .optional),
+        ])
```

Comment:
minor: 

I wonder, should it perhaps be `let hasNoDependencies = target.dependencies.filter(notSDKDependencies).isEmpty` - would there be a valid use case of a target with system dependencies only with no source code or build scripts ?

Reply:
That's an interesting edge case for sure. I'm not sure why anyone would do that, but if they did I think we'd probably want a different warning message. `Target has no sources and is only dependent on system frameworks` or something like that.

My gut feeling is that is such a rare case that I'm not sure if it's worth calling out though. It's possible that someone may have created a fresh target that way and they're on their way to adding sources, so the message would be more nuisance than help.